deploy/helm/templates/metrics-discovery-deployment.yaml

```diff
@@ -0,0 +1,182 @@
+    metadata:
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/metrics-discovery-config-map.yaml") . | sha256sum }}
+        checksum/config_common_env: {{ include (print $.Template.BasePath "/common-env-config-map.yaml") . | sha256sum }}
+        checksum/values: {{ toJson .Values | sha256sum }}
+{{- if .Values.aws_fargate.discovery.telemetry.metrics.enabled }}
+        prometheus.io/scrape: "true"
+        prometheus.io/port: {{ (split ":" .Values.aws_fargate.discovery.telemetry.metrics.address)._1 | quote }}
+        prometheus.io/path: "/metrics"
+        prometheus.io/scheme: "http"
+{{- end}}
+      labels:
+        app.kubernetes.io/name: swo-k8s-collector
+{{ include "common.labels" . | indent 8 }}
+        app: {{ include "common.fullname" (tuple . "-metrics") }}
+{{ include "common.pod-labels" . | indent 8 }}
+    spec:
+      terminationGracePeriodSeconds: {{ .Values.otel.metrics.terminationGracePeriodSeconds }}
+      serviceAccountName: {{ include "common.fullname" . }}
+      securityContext: {}
+      {{- if .Values.aws_fargate.discovery.nodeSelector }}
+      nodeSelector:
+      {{ toYaml .Values.aws_fargate.discovery.nodeSelector | nindent 8 }}
+      {{- end }}
+      {{- if .Values.aws_fargate.discovery.tolerations }}
+      tolerations:
+        {{ toYaml .Values.aws_fargate.discovery.tolerations | nindent 8 }}
+      {{- end }}
+      affinity:
+      {{- if .Values.aws_fargate.discovery.affinity }}
+        {{- toYaml .Values.aws_fargate.discovery.affinity | nindent 8 }}
+      {{- else }}
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: kubernetes.io/arch
+                operator: In
+                values:
+                - amd64
+                - arm64
+              - key: kubernetes.io/os
+                operator: In
+                values:
+                - linux
+      {{- end }}
+      {{- if .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml .Values.imagePullSecrets | nindent 8 }}
+      {{- end }}
+      {{- if or (and .Values.otel.metrics.prometheus_check .Values.otel.metrics.prometheus.url) (include "isSwiEndpointCheckEnabled" .) }}
+      initContainers:
+        {{- if and .Values.otel.metrics.prometheus_check .Values.otel.metrics.prometheus.url }}
+        - name: prometheus-check
+          image: "{{ include "common.image" (tuple . .Values.otel.init_images "busy_box") }}"
+          imagePullPolicy: {{ .Values.otel.init_images.busy_box.pullPolicy }}
+          command: ['sh', '-c', 'until $(wget --spider -nv $PROMETHEUS_URL/federate?); do echo waiting on prometheus; sleep 1; done && echo "Prometheus is available"']
+          envFrom:
+            - configMapRef:
+                name: {{ include "common.fullname" (tuple . "-metrics-env-config") }}
+        {{- end }}
+        {{- if include "isSwiEndpointCheckEnabled" . }}
```

Comment:
We don't use support Prometheus server for discovery. And the `$PROMETHEUS_URL` variable should not even be initialized. I think this does not belong here.